internal/server/engine.go

```diff
@@ -1,18 +1,23 @@
 package server
 
 import (
+	"log"
+
 	"github.com/dhis2-sre/im-manager/internal/middleware"
 	"github.com/dhis2-sre/im-manager/pkg/health"
 	"github.com/gin-contrib/cors"
 	"github.com/gin-gonic/gin"
 	redocMiddleware "github.com/go-openapi/runtime/middleware"
 )
 
-func GetEngine(basePath string) *gin.Engine {
+func GetEngine(basePath string, allowedOrigin string) *gin.Engine {
 	r := gin.Default()
 
 	corsConfig := cors.DefaultConfig()
-	corsConfig.AllowAllOrigins = true
+	// TODO: Allow multiple origins?
+	// Without specifying origin, secure cookies won't work
+	log.Println("Allowed origin:", allowedOrigin)
+	corsConfig.AllowOrigins = []string{allowedOrigin, "http://localhost:5173", "http://localhost:3000"}
```

Comment:
There are some issues here which I'm not entirely sure how to deal with
1. We shouldn't hard code the two localhost url's
2. It's not going to work on http anyway. If we're using http the cookie should be marked as unsecure